crates/provider/src/shared/provider.rs

```diff
@@ -0,0 +1,63 @@
+use std::error::Error;
+
+use async_trait::async_trait;
+
+use super::types::{NativeRunCommandOptions, PodDef, RunCommandResponse};
+
+#[async_trait]
+#[allow(non_upper_case_globals)]
+pub trait Provider {
+    async fn create_namespace(&mut self) -> Result<(), Box<dyn Error>>;
+    fn get_node_ip(&self) -> Result<String, Box<dyn Error>>;
+    fn get_port_mapping(&mut self, port: u16, pod_name: String) -> Result<u16, Box<dyn Error>>;
+    fn get_node_info(&mut self, pod_name: String) -> Result<(String, u16), Box<dyn Error>>;
```

Comment:
I would recommend to add a type aliases to make the code clearer:

```rust
type Port = u16;
```
and it can be relevant to use ```IpAddr``` type from std